docs/src/prototypes/chatMessageWithPopover/Popover.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+  Menu,
+  Popup,
+  toolbarBehavior,
+  popupFocusTrapBehavior,
+  createComponent,
+  ComponentSlotStyle,
+  ComponentVariablesInput,
+  toolbarButtonBehavior,
+} from '@stardust-ui/react'
+import { ReactChildren } from 'types/utils'
+import * as React from 'react'
+import * as cx from 'classnames'
+
+export interface PopoverProps {
+  className?: string
+}
+
+interface PopoverState {
+  focused: boolean
+  popupOpened: boolean
+}
+
+class Popover extends React.Component<PopoverProps, PopoverState> {
+  state = {
+    focused: false,
+    popupOpened: false,
+  }
+
+  changeFocusState = (isFocused: boolean) => {
+    this.state.focused !== isFocused && this.setState({ focused: isFocused })
+  }
+
+  handleFocus = () => {
+    this.changeFocusState(true)
+  }
+
+  handleBlur = e => {
+    // if e.relatedTarget === null, so the click was outside this container
+    if (!this.state.popupOpened || e.relatedTarget === null) {
+      const shouldPreserveFocusState = e.currentTarget.contains(e.relatedTarget)
+      this.changeFocusState(shouldPreserveFocusState)
+    } else {
+      e.stopPropagation()
+    }
+  }
+
+  handleMenuClick = () => {
+    // close popup when other MenuItem clicked, but the event propagation was stopped
+    this.state.popupOpened && this.setState({ popupOpened: false })
+  }
+
+  menuStyles = ({ theme: { siteVariables } }) => ({
+    transition: 'opacity 0.2s',
+    position: 'absolute',
+    top: '-20px',
+    right: '5px',
+    background: siteVariables.white,
+    boxShadow: '0px 2px 4px #ddd',
+    borderRadius: '.3rem',
+    opacity: 0,
+
+    '& .smile-emoji': {
+      display: 'none',
+    },
+
+    '&.focused .smile-emoji': {
+      display: 'flex',
+    },
+
+    '&:hover .smile-emoji': {
+      display: 'flex',
+    },
+  })
+
+  render() {
+    return (
+      <Menu
+        styles={this.menuStyles}
+        iconOnly
+        className={cx(this.props.className, this.state.focused ? 'focused' : '')}
+        items={[
```

Comment:
👍 I see now that you have already implemented this component to respect user's input on the className. See me previous comment for refactoring the other component in the same way.